de-lib/src/main/java/org/iplantc/de/apps/client/views/details/AppDetailsViewImpl.java

```diff
@@ -230,24 +235,61 @@ public void setBaseDebugId(final String baseId) {
             url.addClickHandler(new ClickHandler() {
                 @Override
                 public void onClick(ClickEvent event) {
-                    IPlantPromptDialog ipd = new IPlantPromptDialog(appearance.appUrl(),
-                                                                    1024,
-                                                                    GWT.getHostPageBaseURL()
-                                                                            + "?type="
-                                                                            + DesktopPresenterImpl.TypeQueryValues.APPS
-                                                                            + "&app-id=" + app.getId(),
-                                                                    null);
+                    Dialog ipd = new Dialog();
+                    ipd.setModal(true);
+                    ipd.setHideOnButtonClick(true);
                     ipd.setHeadingHtml(appearance.copyAppUrl());
                     ipd.setWidth("500px");
                     ipd.setPredefinedButtons(PredefinedButton.OK);
                     ipd.show();
+                    renderCopyTextArea(ipd.getBody().getId(true),
+                                       appearance.copyAppUrl(),
+                                       GWT.getHostPageBaseURL()
+                                                    + "?type="
+                                                    + DesktopPresenterImpl.TypeQueryValues.APPS
+                                                    + "&app-id=" + app.getId());
                 }
             });
         } else {
             helpLink.setVisible(false);
         }
+
+        Scheduler.get().scheduleFinally(new Scheduler.ScheduledCommand() {
+
+            @Override
+            public void execute() {
+                renderToolDetails(toolsContainer.getId(),
+                                  getAppDetailsAppearanceShim(appearance),
+                                  AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(app.getTools().get(0))));
```

Comment:
Fortunately a [Splittable is a JavaScriptObject](http://www.gwtproject.org/doc/latest/DevGuideAutoBeans.html#autobeancodex) when passed to the React components!